src/rust/ide/src/view/node_editor.rs

```diff
@@ -127,6 +127,11 @@ impl GraphEditorIntegratedWithController {
     pub fn graph_editor(&self) -> GraphEditor {
         self.model.editor.clone_ref()
     }
+
+    /// Get the controller associated with this graph editor.
+    pub fn controller(&self) -> controller::ExecutedGraph {
+        self.model.controller.clone_ref()
+    }
```

Comment:
I wonder if we could return reference here? Are there any downsides of doing that?